src/main/java/com/dbteam/bot/MoneySplittingBot.java

```diff
@@ -92,16 +92,58 @@ public Ability StartAbility() {
                 .forEach(silent::execute);
         };
 
-        Ability.AbilityBuilder builder = Ability.builder()
-                .name("start")
+        return Ability.builder()
+                .name(Command.START.getValue())
                 .privacy(PUBLIC)
                 .locality(USER)
                 .input(0)
                 .info("I am MoneySplittingBot!")
-                .action(consumer);
+                .action(consumer)
+                .build();
+    }
 
-        return builder.build();
+    public Ability AddPaymentAbility() {
+
+        Consumer<MessageContext> primaryAction = ctx -> {
+            CommandHandler handler = commandHandlerFactory.getHandler(Command.ADD_PAYMENT);
+            handler.primaryAction(ctx.update())
+                    .forEach(silent::execute);
+        };
+
+        Consumer<Update> secondaryConsumer = upd -> {
+            CommandHandler handler = commandHandlerFactory.getHandler(Command.ADD_PAYMENT);
+            handler.secondaryAction(upd)
+                    .forEach(silent::execute);
+        };
 
+        Predicate<Update> condition = upd -> {
+            try {
+                String tempState = stateService
+                        .getUserGroupChatState(
+                                upd.getMessage().getFrom().getUserName(),
+                                upd.getMessage().getChatId());
+                if(tempState != null) {
+                    return tempState.startsWith(Command.ADD_PAYMENT.getValue());
+                } else {
+                    return false;
+                }
+            } catch (PersonNotFoundException e) {
+                e.printStackTrace();
+                return false;
+            }
+        };
+
+        return Ability.builder()
+                .name(Command.ADD_PAYMENT.getValue().toLowerCase())
+                .privacy(PUBLIC)
+                .locality(ALL)        //should this ability be available in userChat with bot? Or only in group (as I did)
```

Comment:
For now it must only be in group chat because it uses chatId of the message in update to get groupChatState of user from database, so if someone does /addpayment in chat with bot, the latter would not know to what group to refer. Later we can modify the ability so it will be usable in both chats.